theseus/__init__.py

```diff
@@ -7,11 +7,15 @@
     CostFunction,
     CostWeight,
     DiagonalCostWeight,
-    AutoDiffCostFunction,
     Objective,
     ScaleCostWeight,
     Variable,
     Vectorize,
+    Loss,
```

Comment:
Have we settled on Loss or are there any better alternatives? Loss good lead to some confusion of loss in the ML context.